File: 1922-count-good-numbers/1922-count-good-numbers.py

class Solution:
    def countGoodNumbers(self, n: int) -> int:
        n=n-1
        m=(10 ** 9 +7)

        if n % 2==0:
            even = n//2 +1
            odd = n//2
        else:
            even=n//2+1
            odd=n//2 +1
        
        evenpow=pow(5,even,m)
        oddpow=pow(4,odd,m)

        return (evenpow * oddpow) % m

File: 1922-count-good-numbers/1922-count-good-numbers.cpp
class Solution {
    const int mode=1e9+7;
    long long power(int a,long long b){
        if(b==0) return 1;

        long long half=power(a,b/2);

        long long result=(half *half )% mode;

        if(b&1){

             result= ( (result*a) %mode);
        }
        return result;
    }
public:
    int countGoodNumbers(long long n) {
        return (power(4,n/2) *power(5,(n+1)/2) % mode);
        
    }
};